src/main/java/org/folio/qm/util/DIEventUtils.java

```diff
@@ -14,28 +14,50 @@
 @UtilityClass
 public class DIEventUtils {
 
-  public static Optional<UUID> extractInstanceId(DataImportEventPayload data, ObjectMapper objectMapper) {
-    return extractRecordId(data, "INSTANCE", objectMapper);
+  public static Optional<UUID> extractExternalId(DataImportEventPayload data, ObjectMapper mapper) {
+    return extractRecordId(data, FolioRecord.INSTANCE, mapper)
+      .or(() -> extractRecordId(data, FolioRecord.HOLDINGS, mapper));
   }
 
-  public static Optional<UUID> extractMarcBibId(DataImportEventPayload data, ObjectMapper objectMapper) {
-    return extractRecordId(data, "MARC_BIBLIOGRAPHIC", objectMapper);
+  public static Optional<UUID> extractMarcId(DataImportEventPayload data, ObjectMapper mapper) {
+    return extractRecordId(data, FolioRecord.MARC_BIBLIOGRAPHIC, mapper)
+      .or(() -> extractRecordId(data, FolioRecord.MARC_HOLDINGS, mapper))
+      .or(() -> extractRecordId(data, FolioRecord.MARC_AUTHORITY, mapper));
   }
 
   public static Optional<String> extractErrorMessage(DataImportEventPayload data) {
     return Optional.ofNullable(data.getContext().get("ERROR"));
   }
 
-  public static Optional<UUID> extractRecordId(DataImportEventPayload data, String recordName, ObjectMapper objectMapper) {
-    return Optional.ofNullable(data.getContext().get(recordName))
-      .map(instancePayload -> {
+  public static Optional<UUID> extractRecordId(DataImportEventPayload data, FolioRecord folioRecord, ObjectMapper mapper) {
+    return Optional.ofNullable(data.getContext().get(folioRecord.getValue()))
+      .map(recordInJson -> {
         try {
-          var idNode = objectMapper.readTree(instancePayload).get("id");
+          var idNode = mapper.readTree(recordInJson).get("id");
           return idNode != null ? UUID.fromString(idNode.asText()) : null;
         } catch (JsonProcessingException e) {
           log.info("Failed to process json", e);
           throw new IllegalStateException("Failed to process json with message: " + e.getMessage());
         }
       });
   }
+
+  public enum FolioRecord {
```

Comment:
1. maybe be better to move FolioRecord from DIEventUtils?
2. Maybe we can reuse this enum instead of marcFormat.json ? For example: we can change marcFormat to fromat:
`"format":"MARC_HOLDINGS"`

And remove marcFormat.json (or extend by adding new values and remove FolioRecord)